general-concepts/dependencies/text.xml

```diff
@@ -209,28 +209,59 @@ asterisk. Also note that when selecting all versions in a specific
 <body>
 
 <p>
-Sometimes two packages cannot be installed in parallel. This is handled by
-blockers. A softblocker is specified as follows:
+When two packages (package slots, versions) can not be installed
+simultaneously, blockers can be used to expose such a conflict
+to the package manager.
 </p>
 
+<p>
+There are two kinds of blockers: soft blockers and hard blockers.
+</p>
+
+<p>A soft blocker is defined using the following syntax:</p>
+
 <codesample lang="ebuild">
 RDEPEND="!app-misc/foo"
 </codesample>
 
 <p>
-Portage will try to resolve this conflict automatically if possible.
-Sometimes we need to use a hardblocker to ensure correct emerge order.
-Those cannot be resolved by Portage and must be taken care of by the user.
-A hardblocker is specified as follows:
+The package manager will try to resolve this conflict automatically.
+The package blocked by a soft blocker can be uninstalled <e>after</e>
+installing the package blocking it. However, it exempts the common
+files from file collision checks. Soft blockers are usually used
+to solve file collisions between packages and are meaningful only
+in <c>RDEPEND</c>.
+</p>
+
+<warning>
+Soft blockers that are pure <c>DEPEND</c> <e>do not work correctly</e>.
+While Portage seemingly queues the package for removal, it <e>does not</e>
+exempt their contents from file collision checks. Always include your
+soft blockers in <c>RDEPEND</c>!
+</warning>
+
+<p>
+If it is strictly necessary to resolve the blocker before the package
+is built (installed) or if automatic resolution is undesired,
+a hard blocker must be used instead. Hard blockers are not resolved
+automatically, and are expressed using the following syntax:
 </p>
 
 <codesample lang="ebuild">
 RDEPEND="!!app-misc/foo"
 </codesample>
 
 <p>
-Hardblockers always take precedence over softblockers and need at least EAPI=2.
-Also note that blockers are usually <e>runtime</e> rather than buildtime.
+Hard blockers apply accordingly to the dependency type defining them.
+Blockers defined in <c>RDEPEND</c> are enforced as long as the package
+is installed (but do not prevent building binary packages). Blockers
+defined purely in <c>DEPEND</c> are enforced only for building
+the package from source, and may not apply once the package is installed
+or when it is installed from a binary package.
+</p>
+
+<p>
+Hard blocks require EAPI 2 or newer.
```

Comment:
I think soft blockers also change behavior between EAPIs 0-1 and 2+
(or rather: behave as you describe in EAPI 2+)

Reply:
Not exactly — they are not soft blockers but undefined-kind-of-blockers then. The previous version explained that in detail but @ulm requested that I revert to the previous wording which is this.

Reply:
Why not simply put a note that this section applies to EAPI2+ then ? (because everyone should be using something even newer actually)
I don't see any value in documenting something that has been improved/fixed in EAPI2 for a reason.

Reply:
Added.